packages/near-membrane-base/src/environment.ts

```diff
@@ -191,9 +191,9 @@ export class VirtualEnvironment {
             const key = keys[i];
             bluePointer = this.blueCallableGetPropertyValuePointer(bluePointer, key);
             redPointer = this.redCallableGetPropertyValuePointer(redPointer, key);
+            this.redCallableLinkPointers(redPointer, bluePointer);
+            this.blueCallableLinkPointers(bluePointer, redPointer);
         }
-        this.redCallableLinkPointers(redPointer, bluePointer);
-        this.blueCallableLinkPointers(bluePointer, redPointer);
```

Comment:
Avoids looping over the same properties over and over. Instead linking as it walks.